### PR TITLE
Use the same cacher for public key and signing caches

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-346

**Release Note**

```release-note
Use the configured cacher for public key and signing key caches
```

/assign @jeremyfaller

### CODE REVIEW

File: pkg/controller/jwks/jwks.go [hunk 1]
```diff
@@ -52,8 +52,25 @@ type Controller struct {
 func (c *Controller) HandleIndex() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
+
```
Comment (sethvargo) [inline pkg/controller/jwks/jwks.go:55]: @jeremyfaller can you take a look at this? There was a subtle bug that would result in a deadlock if the same cacher was given to the public key cache and the jwk builder. I think I fixed it, but would appreciate a second look.

**Review by sethvargo — COMMENTED** [review]
/hold to test other signers

File: pkg/controller/jwks/jwks.go (hunk 1 above)
Comment (jeremyfaller) [reply pkg/controller/jwks/jwks.go:55]: Not sure I completely see the deadlock, would be great to have a test to prove it's fixed. In any case:

AFA the JWK stuff, this separation looks like it makes sense.

**Review by jeremyfaller — APPROVED** [review]

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**